### ISSUE TITLE
Can't pass data to component from beforeRouteEnter only after component is initialized

### ISSUE THREAD

**Issue by wizardnet972** [vuejs/vue-router#2857]
### Version
3.0.7

### Reproduction link
[https://codesandbox.io/s/vue-typescript-example-cb9o0](https://codesandbox.io/s/vue-typescript-example-cb9o0)

### Steps to reproduce
Open the reproduction.
open the console.
Then click on the link: go to page.

check the console log.

### What is expected?
console log object

### What is actually happening?
console log undefined.

---
I think because next function happens after the component is initialized and in setTimeout((), 16) I can't be able to pass to its props before the component is executed.

I think right now is weird behavior for the router. its miss the expected guard/resolve behavior.

So this is why I got undefined at the getter function.

Any idea how to solve this issue?

**Comment by posva** [comment]
Duplicate of https://github.com/vuejs/vue-router/issues/1144
